File: apple/Murmur/Components/Card.swift
import SwiftUI

/// The rounded panel every screen is built out of — the phone's answer to the desktop's
/// `.card` class (`src/index.css`): a slightly raised surface, 14 pt corners, no border.
///
/// It uses `secondarySystemGroupedBackground` rather than a fixed colour so it sits correctly
/// on the grouped background in both light and dark mode without a second palette.
struct Card<Content: View>: View {
    var padding: CGFloat = 16
    @ViewBuilder var content: Content

    var body: some View {
        content
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding(padding)
            .background(Color(.secondarySystemGroupedBackground), in: RoundedRectangle(cornerRadius: 14, style: .continuous))
    }
}

extension View {
    /// The screen background behind cards: the same grouped grey `List` uses, so a screen
    /// built from cards and a screen built from a `Form` look like one app.
    func murmurScreenBackground() -> some View {
        background(Color(.systemGroupedBackground).ignoresSafeArea())
    }
}

/// A small all-caps label above a group, matching the desktop's `.section-label`.
struct SectionLabel: View {
    let text: String

    init(_ text: String) { self.text = text }

    var body: some View {
        Text(text.uppercased())
            .font(.caption2.weight(.semibold))
            .kerning(0.6)
            .foregroundStyle(.secondary)
            .frame(maxWidth: .infinity, alignment: .leading)
    }
}

File: android/app/src/main/java/com/murmur/app/ui/components/Card.kt
package com.murmur.app.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ColumnScope
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * The rounded panel every screen is built out of — the phone's answer to the desktop's `.card`
 * class (`src/index.css`), and the twin of iOS's `Card`: a raised surface, 14 dp corners, no
 * border.
 */
@Composable
fun MurmurCard(
    modifier: Modifier = Modifier,
    padding: Int = 16,
    content: @Composable ColumnScope.() -> Unit,
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(14.dp))
            .background(MaterialTheme.colorScheme.surface)
            .padding(padding.dp),
        content = content,
    )
}

/** A small all-caps label above a group, matching the desktop's `.section-label`. */
@Composable
fun SectionLabel(text: String, modifier: Modifier = Modifier) {
    Text(
        text = text.uppercase(),
        style = MaterialTheme.typography.labelSmall,
        fontWeight = FontWeight.SemiBold,
        letterSpacing = 0.6.sp,
        color = MaterialTheme.colorScheme.onSurfaceVariant,
        modifier = modifier.fillMaxWidth(),
    )
}
